From efdf5f8cc6af88b3c25073b0d7047b7a238cb24b Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Mon, 16 Aug 2021 11:39:43 +0900
Subject: [PATCH 8/8] [ozone/wayland] Force set pointer focus when the window
 gets touch focus

It intends to enable wayland input methods such as weston-keyboard
without real keyboard & mouse. Without this patch, IMs cannot be
activated on content area when only touch panel is connected.

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 ui/ozone/platform/wayland/host/wayland_event_source.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ui/ozone/platform/wayland/host/wayland_event_source.cc b/ui/ozone/platform/wayland/host/wayland_event_source.cc
index 6e23105258a31..1ed349e5bd1b4 100644
--- a/ui/ozone/platform/wayland/host/wayland_event_source.cc
+++ b/ui/ozone/platform/wayland/host/wayland_event_source.cc
@@ -434,6 +434,8 @@ void WaylandEventSource::HandleTouchFocusChange(WaylandWindow* window,
   DCHECK(window);
   bool actual_focus = id ? !ShouldUnsetTouchFocus(window, id.value()) : focused;
   window->set_touch_focus(actual_focus);
+  if (actual_focus)
+    HandlePointerFocusChange(window);
 }
 
 // Focus must not be unset if there is another touch point within |window|.
-- 
2.30.2

